nix/nixpkgs.nix

```diff
@@ -1,8 +1,16 @@
 {}:
 
 let
-  nixpkgs = (import ./sources.nix).nixpkgs;
-  overlay = (import ./sources.nix).nixpkgs-mozilla;
+  sources = import ./sources.nix;
+  nixpkgs = sources.nixpkgs;
+  overlay = sources.nixpkgs-mozilla;
+  naersk = sources.naersk;
   mdsh    = (import ./mdsh.nix);
 in
-import nixpkgs { overlays = [ (import overlay) mdsh ]; }
+import nixpkgs { overlays =
+  [ (import overlay)
+    (_: _: { inherit sources; })
```

Comment:
I'm afraid my nix-foo is failing me here. Why do we need `(_: _: { inherit sources; })` ?